crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -256,19 +276,34 @@ impl<'a> ReflectStruct<'a> {
         &self.meta
     }
 
-    /// Get an iterator over the active fields.
-    pub fn active_fields(&self) -> impl Iterator<Item = &StructField<'a>> {
-        self.fields.iter().filter(|field| !field.attrs.ignore)
+    /// Get an iterator over the fields satisfying the given predicate
+    fn fields_with<F: FnMut(&&StructField) -> bool>(
+        &self,
+        predicate: F,
+    ) -> impl Iterator<Item = &StructField<'a>> {
+        self.fields.iter().filter(predicate)
     }
 
-    /// Get an iterator over the ignored fields.
-    pub fn ignored_fields(&self) -> impl Iterator<Item = &StructField<'a>> {
-        self.fields.iter().filter(|field| field.attrs.ignore)
+    /// Get a collection of all field types satisfying the given predicate
+    fn types_with<F: FnMut(&&StructField) -> bool>(&self, predicate: F) -> Vec<syn::Type> {
```

Comment:
That was just for internal convenience, filter operates over references and since `iter` generates an iterator of references, your predicate ends up being over double references, I should be able to change this though to look better externally